src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -373,14 +373,18 @@ static int add_partition_swap(DissectedPartition *p) {
         if (!arg_swap_enabled)
                 return 0;
 
-        /* Disable the swap auto logic if at least one swap is defined in /etc/fstab, see #6192. */
-        r = fstab_has_fstype("swap");
-        if (r < 0)
-                return log_error_errno(r, "Failed to parse fstab: %m");
-        if (r > 0) {
-                log_debug("swap specified in fstab, ignoring.");
-                return 0;
+        /* Prioritize fstab for honoring custom swap options, see #6192. */
+        r = fstab_has_node(p->node);
```

Comment:
hmm, i still don't like this. swap is quite often in a luks envelope, which means this check really isn't sufficient, because at the moment gpt-auto-generator while the partition exists it is quite likely not unlocked yet...